File: scala/sdktest/OutputResetPasswordEntityTest.scala
// Generated basic-flow test for the output_reset_password entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped OutputResetPasswordTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsuserbackofficesdk.core.{Helpers, BluefinTecsUserBackofficeSDK}
import voxgig.bluefintecsuserbackofficesdk.utility.struct.Struct

object OutputResetPasswordEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("output_reset_password.instance") {
      val testsdk = BluefinTecsUserBackofficeSDK.testSDK()
      val ent = testsdk.outputResetPassword(null)
      rep.check("output_reset_password.instance", ent != null, "expected non-null output_reset_password entity")
    }

    rep.scope("output_reset_password.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/output_reset_password/OutputResetPasswordTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsUserBackofficeSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("output_reset_password01", "OUTPUT_RESET_PASSWORD01")
      idmap.put("output_reset_password02", "OUTPUT_RESET_PASSWORD02")
      idmap.put("output_reset_password03", "OUTPUT_RESET_PASSWORD03")
      val now = System.currentTimeMillis()

      // CREATE
      val outputResetPasswordRef01Ent = client.outputResetPassword(null)
      var outputResetPasswordRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.output_reset_password"), "output_reset_password_ref01"))
      val outputResetPasswordRef01DataResult = outputResetPasswordRef01Ent.create(outputResetPasswordRef01Data, null)
      outputResetPasswordRef01Data = Helpers.toMapAny(outputResetPasswordRef01DataResult)
      rep.check("output_reset_password.create.map", outputResetPasswordRef01Data != null, "expected create result to be a map")
    }
  }
}

File: js/test/entity/output_reset_password/OutputResetPasswordEntity.test.js

const envlocal = __dirname + '/../../../.env.local'
require('dotenv').config({ quiet: true, path: [envlocal] })

const Path = require('node:path')
const Fs = require('node:fs')

const { test, describe } = require('node:test')
const assert = require('node:assert')


const { BluefinTecsUserBackofficeSDK, BaseFeature, stdutil, config } = require('../../..')

const {
  envOverride,
  makeCtrl,
  makeMatch,
  makeReqdata,
  makeStepData,
  makeValid,
} = require('../../utility')


describe('OutputResetPasswordEntity', async () => {

  test('instance', async () => {
    const testsdk = BluefinTecsUserBackofficeSDK.test()
    const ent = testsdk.OutputResetPassword()
    assert(null != ent)
  })


  // Feature #4: the entity `stream(action, ...)` method runs the op pipeline
  // and returns an async iterator over result items. With the streaming
  // feature active it yields the feature's incremental output; otherwise it
  // falls back to the materialised list so `stream` always yields.
  test('stream', async () => {
    const seed = {
      entity: {
        output_reset_password: { s1: { id: 's1' }, s2: { id: 's2' }, s3: { id: 's3' } }
      }
    }

    // Fallback: streaming inactive -> yields the materialised list items.
    const base = BluefinTecsUserBackofficeSDK.test(seed)
    const seen = []
    for await (const item of base.OutputResetPassword().stream('list')) {
      seen.push(item)
    }
    assert.equal(seen.length, 3)

    // Inbound: streaming active -> yields each item from the feature iterator.
    if (config.feature && config.feature.streaming) {
      const sdk = BluefinTecsUserBackofficeSDK.test(seed, { feature: { streaming: { active: true } } })
      const got = []
      for await (const item of sdk.OutputResetPassword().stream('list')) {
        if (Array.isArray(item)) { got.push(...item) } else { got.push(item) }
      }
      assert.equal(got.length, 3)
    }
  })


  test('basic', async () => {

    const setup = basicSetup()
    const client = setup.client
    const struct = setup.struct

    const isempty = struct.isempty
    const select = struct.select


    // CREATE
    const output_reset_password_ref01_ent = client.OutputResetPassword()
    let output_reset_password_ref01_data = setup.data.new.output_reset_password['output_reset_password_ref01']

    output_reset_password_ref01_data = await output_reset_password_ref01_ent.create(output_reset_password_ref01_data)
    assert(null != output_reset_password_ref01_data)


  })
})



function basicSetup(extra) {
  // TODO: fix test def options
  const options = {} // null

  // TODO: needs test utility to resolve path
  const entityDataFile =
    Path.resolve(__dirname,
      '../../../../.sdk/test/entity/output_reset_password/OutputResetPasswordTestData.json')

  // TODO: file ready util needed?
  const entityDataSource = Fs.readFileSync(entityDataFile).toString('utf8')

  // TODO: need a xlang JSON parse utility in voxgig/struct with better error msgs
  const entityData = JSON.parse(entityDataSource)

  options.entity = entityData.existing

  let client = BluefinTecsUserBackofficeSDK.test(options, extra)
  const struct = client.utility().struct
  const merge = struct.merge
  const transform = struct.transform

  let idmap = transform(
    ['output_reset_password01','output_reset_password02','output_reset_password03'],
    {
      '`$PACK`': ['', {
        '`$KEY`': '`$COPY`',
        '`$VAL`': ['`$FORMAT`', 'upper', '`$COPY`']
      }]
    })

  const env = envOverride({
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_RESET_PASSWORD_ENTID': idmap,
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_LIVE': 'FALSE',
    'BLUEFIN_TECS_USER_BACKOFFICE_TEST_EXPLAIN': 'FALSE',
    'BLUEFIN_TECS_USER_BACKOFFICE_APIKEY': 'NONE',
  })

  idmap = env['BLUEFIN_TECS_USER_BACKOFFICE_TEST_OUTPUT_RESET_PASSWORD_ENTID']

  if ('TRUE' === env.BLUEFIN_TECS_USER_BACKOFFICE_TEST_LIVE) {
    client = new BluefinTecsUserBackofficeSDK(merge([
      {
        apikey: env.BLUEFIN_TECS_USER_BACKOFFICE_APIKEY,
      },
      extra
    ]))
  }

  const setup = {
    idmap,
    env,
    options,
    client,
    struct,
    data: entityData,
    explain: 'TRUE' === env.BLUEFIN_TECS_USER_BACKOFFICE_TEST_EXPLAIN,
    now: Date.now(),
  }

  return setup
}
  
